src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -13,7 +13,34 @@ internal abstract class TagsList : ITags
 
         public List<KeyValuePair<string, double>> Metrics => Volatile.Read(ref _metrics);
 
-        public List<KeyValuePair<string, string>> Tags => Volatile.Read(ref _tags);
+        public List<KeyValuePair<string, string>> GetAll()
```

Comment:
**blocker bug:**
I think this implementation breaks `ToString()` , `SerializeTo()` and maybe even other stuff.